GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -814,6 +839,12 @@ private void ShowLoading(bool sync = true)
         }
 
         public void ForceRefreshRevisions()
+        {
+            Dictionary<ObjectId, string> filePathByObject = null;
+            ForceRefreshRevisions(ref filePathByObject);
```

Comment:
This doesn't make sense to me

Reply:
`BuildFilter()` is needed for path arguments both from FileHistory and  Advanced filter (this PR removes the separate path filter handling too).
The routine finds the previous filenames and the name(s) in previous commits, mostly used for presentation. The latter part is only needed in FileHistory.
With FileHistory it is possible to "insert" `BuildFilter()` before RevisionGridControl handling, not possible with changes in Advanced filter.
So `filePathByObject` is only needed by FileHistory.

Note that `BuildFilter()` also handles quoting of path arguments to handle user input in Advanced filter better.

There are alternatives to produce `filePathByObject`, some examples:
* Set `filePathByObject` to null when not needed (other than FileHistory requesting) and update the variable if not null. - What is in the PR now, in my view the simplest with the least amount of code.
* Always update the variable, referred by a ref. (in the snippet above, null would be changed to new()) - Similar, just requires a few more CPU cycles.
* Always update the variable, let FileHistory ask for it afterwards. - Requires a few more CPU cycles and yet another property but maybe the overloaded `ForceRefreshRevisions()` is not needed.
* Make producing `filePathByObject` a RevisionGrid configurable, let FileHistory ask for it afterwards. - Similar to above.
* Move the handling of `filePathByObject` to FileHistory, requiring a third git-log invocation. - Straightforward but can add seconds to the FileHistory opening time.

Reply:
An alternative implementation in e58fa066b09f443de7826f675c2b847fb05f7aff, is that OK?
(similar to alternative 4 above)

Reply:
> An alternative implementation in [e58fa06](https://github.com/gitextensions/gitextensions/commit/e58fa066b09f443de7826f675c2b847fb05f7aff), is that OK?

This looks better 👍

Reply:
Why set this as draft?